project4/name_server/tasks/CMakeLists.txt

```diff
@@ -0,0 +1,2 @@
+add_library(name_server_tasks_lib console_task.cpp bootstrap_task.cpp nameserver_task.cpp ../nameserver.cpp ../bootstrap.cpp)
```

Comment:
`nameserver.cpp` and `bootstrap.cpp` should not be included here.

Reply:
It complains without having it.

Reply:
This circular dependency suggests that there is some larger design issue. It's hard to say without seeing the actual errors, but my guess is that this is caused by instantiating `Nameserver` and `Bootstrap` classes within the tasks. The easiest way to solve this problem would be to pass in a `shared_ptr` (or `unique_ptr`) of the `Nameserver` or `Bootstrap` instance that you want to use in the constructor of the relevant task. This also has the benefit of reducing coupling and making your code more testable.